### PR TITLE
Serialize

### PR DESCRIPTION
**Pull request by anotherkowski**
User, List, and Item serializers working

### CODE REVIEW

File: app/serializers/topic_serializer.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class TopicSerializer < ActiveModel::Serializer
+  attributes :id
+  has_many :items
```
Comment (bobbilee19) [inline app/serializers/topic_serializer.rb:3]: there are no relationships in serializers :( these are done in the models

File: app/serializers/user_serializer.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class UserSerializer < ActiveModel::Serializer
+  attributes :id, :created_at, :name, :email, :password
+  has_many :lists
+  has_many :items
+
+  # Delegate the practical definition of `full_name` to
```
Comment (bobbilee19) [inline app/serializers/user_serializer.rb]: yeah same here and the methods here should be in the models

for example...
```
class User < ApplicationRecord
  has_many :lists, dependent: :destroy
  validates :name, presence: true
  validates :email, presence: true
  validates :password_digest, presence: true
end
```

File: app/serializers/user_serializer.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  # the User model, where it belongs, rather than
+  # (re)defining it here.
+  def name
+    object.name
```
Comment (bobbilee19) [inline app/serializers/user_serializer.rb]: remove this

File: app/serializers/user_serializer.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  end
+
+  def created_at
+    object.created_at.strftime('%B %d, %Y')
```
Comment (bobbilee19) [inline app/serializers/user_serializer.rb]: remove this as well

File: app/serializers/list_serializer.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+class ListSerializer < ActiveModel::Serializer
+  attributes :id, :title, :user_id, :public, :created_at
+  has_many :items
```
Comment (bobbilee19) [inline app/serializers/list_serializer.rb]: yeah no relationships here either...

ur model deals w all this

and created_at is smething that is autogenerated remember when a new object is created

so ur model for list shold look like so..

```
class List < ApplicationRecord
  belongs_to :user
  has_many :items, dependent: :destroy
  validates :title, presence: true
end
```

File: app/serializers/topic_serializer.rb (hunk 1 above)
Comment (anotherkowski) [reply app/serializers/topic_serializer.rb:3]: Got it. Removing that then.

File: app/serializers/insecure_user_serializer.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+class InsecureUserSerializer < ActiveModel::Serializer
+  attributes :id, :email, :password, :name
+
+  def name
```
Comment (bobbilee19) [inline app/serializers/insecure_user_serializer.rb:4]: no :(

File: app/serializers/list_serializer.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+class ListSerializer < ActiveModel::Serializer
+  attributes :id, :title, :user_id, :public, :created_at
+
+  def created_at
+    object.created_at.strftime('%B %d, %Y')
```
Comment (bobbilee19) [inline app/serializers/list_serializer.rb:5]: no :(

File: app/serializers/item_serializer.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+class ItemSerializer < ActiveModel::Serializer
+  attributes :id, :title, :list_id, :user_id, :created_at
+
+  def created_at
+    object.created_at.strftime('%B %d, %Y')
```
Comment (bobbilee19) [inline app/serializers/item_serializer.rb:5]: no :(

File: app/serializers/topic_serializer.rb (hunk 1 above)
Comment (bobbilee19) [reply app/serializers/topic_serializer.rb:3]: Looks like u merged in all those mistakes that i mentioned :(

**Comment by anotherkowski** [comment]
In the Bloc guide, it told me to include those extra methods: https://www.bloc.io/resources/serializers. If they're not necessary, I can delete them all.

**Pull request merged**